pkg/cloudclient/aws/private.go

```diff
@@ -312,6 +308,144 @@ func getMasterNodeSubnets(kclient client.Client) (map[string]string, error) {
+	if err != nil {
+		log.Error(err, "Cannot decode AWS Provider Spec")
+		return nil, err
+	}
+
+	// Get the subnet the master machine is in
+	initalSubnet := machineProviderSpec.Subnet.ID
+
+	// Look up the subnet metadata
+	describeSubnetOutput, err := c.ec2Client.DescribeSubnets(&ec2.DescribeSubnetsInput{SubnetIds: []*string{initalSubnet}})
+
+	// Extract the VPC ID from the subnet metadata
+	targetVPC := describeSubnetOutput.Subnets[0].VpcId
+
+	// List all subnets in the VPC
+	allSubnets, err := c.getAllSubnetsInVPC(*targetVPC)
+
+	// List all route tables associated with the VPC
+	routeTables, err := c.getAllRouteTablesInVPC(*targetVPC)
+
+	for _, subnet := range allSubnets {
+		isPublic, err := isSubnetPublic(routeTables, *subnet.SubnetId)
+
+		if err != nil {
+			log.Error(err, "Error while determining if subnet is public")
+			return nil, err
+		}
+		if isPublic {
+			publicSubnets = append(publicSubnets, *subnet.SubnetId)
+		}
+	}
+
+	return publicSubnets, nil
+}
+
+func (c *Client) getAllSubnetsInVPC(vpcID string) ([]*ec2.Subnet, error) {
+
+	var subnetIDs []*ec2.Subnet
+	token := aws.String("initString")
+
+	for token != nil {
+		describeSubnetOutput, err := c.ec2Client.DescribeSubnets(&ec2.DescribeSubnetsInput{Filters: []*ec2.Filter{{Name: aws.String("vpc-id"), Values: []*string{aws.String(vpcID)}}}})
+		if err != nil {
+			log.Error(err, "Error while descirbing subnets")
+			return nil, err
+		}
+		for _, subnet := range describeSubnetOutput.Subnets {
+			subnetIDs = append(subnetIDs, subnet)
+		}
+
+		token = describeSubnetOutput.NextToken
+	}
+
+	return subnetIDs, nil
+}
+
+func (c *Client) getAllRouteTablesInVPC(vpcID string) ([]*ec2.RouteTable, error) {
+
+	var routeTables []*ec2.RouteTable
+	token := aws.String("initString")
+
+	for token != nil {
+		describeRouteTablesOutput, err := c.ec2Client.DescribeRouteTables(&ec2.DescribeRouteTablesInput{Filters: []*ec2.Filter{{Name: aws.String("vpc-id"), Values: []*string{aws.String(vpcID)}}}})
+		if err != nil {
+			log.Error(err, "Error while describing route tables")
+			return nil, err
+		}
+		for _, routeTable := range describeRouteTablesOutput.RouteTables {
+			routeTables = append(routeTables, routeTable)
+		}
+
+		token = describeRouteTablesOutput.NextToken
+	}
+
+	return routeTables, nil
+}
+
+func isSubnetPublic(rt []*ec2.RouteTable, subnetID string) (bool, error) {
+	var subnetTable *ec2.RouteTable
+	for _, table := range rt {
+		for _, assoc := range table.Associations {
+			if aws.StringValue(assoc.SubnetId) == subnetID {
+				subnetTable = table
+				break
+			}
+		}
+	}
+
+	if subnetTable == nil {
+		// If there is no explicit association, the subnet will be implicitly
+		// associated with the VPC's main routing table.
+		for _, table := range rt {
+			for _, assoc := range table.Associations {
+				if aws.BoolValue(assoc.Main) {
+					logrus.Debugf("Assuming implicit use of main routing table %s for %s",
+						aws.StringValue(table.RouteTableId), subnetID)
```

Comment:
```suggestion
					log.Info(fmt.Sprintf("Assuming implicit use of main routing table %s for %s",
						aws.StringValue(table.RouteTableId), subnetID))
```

If we swap this out, we can drop the import

Reply:
done